pkg/kdump/config-client.es6

```diff
@@ -0,0 +1,165 @@
+        this.settings = { };
+
+        cockpit.event_target(this);
+
+        this._fileHandle = cockpit.file(filename, { superuser: superuser } );
+        var self = this;
+        this._fileHandle.watch(function(rawContent) {
+            self._parseText(rawContent);
+        });
+    }
+    close() {
+        if (this._fileHandle) {
+            this._fileHandle.remove();
+            this._fileHandle = undefined;
+        }
+    }
+    // wait for data to have been read at least once
+    wait() {
+        return this._dataAvailable.promise();
+    }
+    /* parse lines of the config file
+     * if a line has a valid config key, use that as key
+     * and also store original line, line index, value and whether the line contains a comment
+     * we care about the comment since we don't want to overwrite a user comment when changing a value
+     * e.g. for line "someKey foo # comment"
+     * outputObject["someKey"] = { index: 0, value: "foo", origLine: "someKey foo # comment", hasComment: true }
+     * skipNotify: Don't notify about changes, e.g.to avoid multiple updates when writing a file
+     */
+    _parseText(rawContent, skipNotify) {
+        if (this._dataAvailable)
+            this._dataAvailable.resolve();
+
+        var self = this;
+        // if nothing changed, don't bother parsing the content
+        if (rawContent == self._rawContent)
+            return;
+
+        if (skipNotify === undefined)
+            skipNotify = false;
+
+        self._rawContent = rawContent;
+        // parse the config file
+        self._lines = rawContent.split(/\r?\n/);
+
+        self.settings = { };
+        self._lines.forEach(function(line, index) {
+            let trimmed = line.trimLeft();
+            // if the line is empty or only a comment, skip
+            if (trimmed.indexOf("#") === 0 || trimmed.length === 0)
+                return;
+
+            // we need to have a space between key and value
+            let separatorIndex = trimmed.indexOf(" ");
+            if (separatorIndex === -1)
+                return;
+            let key = trimmed.substring(0, separatorIndex);
+            let value = trimmed.substring(separatorIndex+1).trim();
+
+            // value might have a comment at the end
+            let commentIndex = value.indexOf("#");
+            let comment;
+            if (commentIndex !== -1) {
+                comment = value.substring(commentIndex).trim();
+                value = value.substring(0, commentIndex).trim();
+            }
+            self.settings[key] = {
+                index: index,
+                value: value,
+                origLine: line,
+                comment: comment
+            };
+        });
+
+        // make sure we copy the original keys so we overwrite the correct lines when saving
+        self._originalSettings = { };
+        Object.keys(self.settings).forEach(function(key) {
+            self._originalSettings[key] = cockpit.extend({}, self.settings[key]);
```

Comment:
I had to change this back to using `cockpit.extend` since the unit test for the config client fails when using `Object.assign`